watchdog/src/health.rs

```diff
@@ -117,14 +144,69 @@ mod test {
 
     #[test]
     fn test_median() {
-        assert_eq!(median(vec![]), None);
-        assert_eq!(median(vec![1]), Some(1));
-        assert_eq!(median(vec![2, 1]), Some(1));
-        assert_eq!(median(vec![3, 2, 1]), Some(2));
-        assert_eq!(median(vec![4, 3, 2, 1]), Some(2));
-        assert_eq!(median(vec![5, 4, 3, 2, 1]), Some(3));
-        assert_eq!(median(vec![20, 20, 10, 10]), Some(15));
-        assert_eq!(median(vec![20, 15, 10]), Some(15));
+        assert_eq!(median(&[]), None);
+        assert_eq!(median(&[1]), Some(1));
+        assert_eq!(median(&[2, 1]), Some(1));
+        assert_eq!(median(&[3, 2, 1]), Some(2));
+        assert_eq!(median(&[4, 3, 2, 1]), Some(2));
+        assert_eq!(median(&[5, 4, 3, 2, 1]), Some(3));
+        assert_eq!(median(&[20, 20, 10, 10]), Some(15));
+        assert_eq!(median(&[20, 15, 10]), Some(15));
+    }
+
+    /// Test data for `calculate_height_target`.
+    struct CalculateHeightTargetTestData {
+        heights: &'static [u64],
+        min_explorers: usize,
+        blocks_behind_threshold: i64,
+        blocks_ahead_threshold: i64,
+        expected: Option<u64>,
+    }
+
+    /// Tests `calculate_height_target` with the given test data.
+    fn test_calculate_height_target(params: CalculateHeightTargetTestData) {
+        assert_eq!(
+            calculate_height_target(
+                params.heights,
+                params.min_explorers,
+                params.blocks_behind_threshold,
+                params.blocks_ahead_threshold
+            ),
+            params.expected
+        );
+    }
+
+    #[test]
+    fn test_calculate_height_target_not_enough_explorers() {
+        test_calculate_height_target(CalculateHeightTargetTestData {
+            heights: &[10],
+            min_explorers: 3,
+            blocks_behind_threshold: -1,
+            blocks_ahead_threshold: 1,
+            expected: None,
+        });
+    }
+
+    #[test]
+    fn test_calculate_height_target_explorers_not_in_range() {
+        test_calculate_height_target(CalculateHeightTargetTestData {
+            heights: &[10, 20, 30],
+            min_explorers: 3,
+            blocks_behind_threshold: -1,
+            blocks_ahead_threshold: 1,
+            expected: None,
+        });
+    }
+
+    #[test]
+    fn test_calculate_height_target_explorers_are_in_range() {
+        test_calculate_height_target(CalculateHeightTargetTestData {
+            heights: &[10, 11, 12],
+            min_explorers: 3,
+            blocks_behind_threshold: -1,
+            blocks_ahead_threshold: 1,
+            expected: Some(11),
+        });
```

Comment:
Maybe I will add a couple more of test variations, e.g.:

1. We have explorers only above the threshold that are in the range(lo, hi), and their number is < `min_explorers`
2. We have explorers only above the threshold that are in the range, and their number is >= `min_explorers`
3. We have explorers only below the threshold that are in range, and their number is < `min_explorers`
4. We have explorers only below the threshold that are in range, and their number is >= `min_explorers`

Reply:
Done.